### PR TITLE
[pdal] delete and replace different find modules

### PR DESCRIPTION
**Pull request by Neumann-A**
makes linkage within pdal more robust.

### CODE REVIEW

**Comment by Neumann-A** [comment]
ah this requires probably the geotiff pr #6596

**Comment by vicroms** [comment]
#6596 has been merged to `master` 
I'll trigger a new build.

**Comment by vicroms** [comment]
Results: `pdal` fails on `x86-windows` and `x64-windows`

[Failure logs](https://github.com/microsoft/vcpkg/files/3218770/pdalFailureLogs.zip)

**Comment by Neumann-A** [comment]
Seems like it did not get the memo that geotiff has been updated. Will merge the master later to this pr. The logs clearly shows that it is the proj.lib vs proj target error.

**Comment by vicroms** [comment]
Looks like this time is pdal-c instead of pdal that's failing on x86- and x64- Windows.
 
I'm away from a PC, so I can't get the logs right now.

**Comment by cenit** [comment]
Again... not to complain but I am sure you know that like the geotiff patch this is already present in #6000 since some time...
Isn’t it better to join effort instead of resubmitting same patch in another pr? It just adds work for everybody. For you, for @vicroms and for me, that I have to merge master for nothing

**Comment by Neumann-A** [comment]
>  not to complain but I am sure you know that like the geotiff patch this is already present in #6000 since some time...

I dont know that PR claims to fix LZMA and not geotiff and I am not actively watching all activtiy in vcpkg. I am just adding PRs to someday maybe get #5543 (or #6555) into vcpkg which will generally solve all link issues in regards to debug/release by injecting the correct vcpkg paths if necessary. 

> It just adds work for everybody.

https://github.com/microsoft/vcpkg/pull/6608#issuecomment-495758747

**Comment by Neumann-A** [comment]
Regressions? The Preview CI does not show any.

**Comment by vicroms** [comment]
All failures occurred in Mac OS, which is not yet on Preview CI.

@Rastaban can you provide logs for build [2715856](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_build/results?buildId=2715856)?

|x64-osx         |test|notes
|----------------|----|-----
|discord-rpc     |Pass|
|mpfr            |Fail|Previously skipped, not a regression
|cjson           |Pass|New Port
|gl3w            |Skip|Missing port dependency
|pangolin        |Fail|Previously skipped, not a regression
|tmxparser       |Pass|
|libressl        |Fail|**Regression**
|highfive        |Fail|**Regression**
|glad            |Skip|Missing port dependency
|geogram         |Skip|Missing port dependency
|xxhash          |Fail|**Regression**
|clapack         |Fail|**Regression**
|fizz            |Skip|Missing port dependency
|simpleini       |Fail|**Regression**
|sophus          |Skip|Missing port dependency
|mlpack          |Skip|Missing port dependency
|g2o             |Skip|Missing port dependency
|libopenmpt      |Fail|Previously skipped, not a regression
|ceres           |Skip|Missing port dependency
|ensmallen       |Skip|Missing port dependency
|tinyexif        |Fail|**Regression**
|wangle          |Skip|Missing port dependency
|mpg123          |Pass|
|egl-registry    |Fail|**Regression**
|armadillo       |Fail|**Regression**
|intelrdfpmathlib|Fail|**Regression**
|opengl-registry |Skip|Missing port dependency
|aubio           |Fail|Previously skipped, not a regression
|ptex            |Fail|**Regression**
|libsodium       |Fail|**Regression**
|pbc             |Fail|Previously skipped, not a regression
|ffmpeg          |Pass|
|fastcdr         |Fail|**Regression**
|blaze           |Skip|Missing port dependency
|mpir            |Pass|
|quirc           |Fail|**Regression**
|angle           |Skip|Missing port dependency
|sndfile         |Fail|**Regression**

**Comment by Neumann-A** [comment]
since i am only touching pdal and the only port dependent on pdal is pdal-c there seems to be an error in the CI build

**Comment by vicroms** [comment]
/azp run

That's true, the CI results already looked suspicious once they only appeared on Mac OS.

**Comment by vicroms** [comment]
You're right, the CI results already looked suspicious once they only appeared on Mac OS.

File: ports/pdal/PDALConfig.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/PDALConfig.cmake.in b/PDALConfig.cmake.in
+index a03ef142e..2d06a2937 100644
+--- a/PDALConfig.cmake.in
++++ b/PDALConfig.cmake.in	
+@@ -15,6 +15,8 @@ foreach(_dir @PDAL_CONFIG_LIBRARY_DIRS@)
+   list(APPEND PDAL_LIBRARY_DIRS ${_foo})
+ endforeach(_dir)
+ 
++include(CMakeFindDependencyMacro)
++find_dependency(GeoTIFF 1.3.0)
```
Comment (vicroms) [inline ports/pdal/PDALConfig.patch]: Is it strictly necessary to require this specific version?

Comment (Neumann-A) [reply ports/pdal/PDALConfig.patch]: when i first tested it, a version was required. This seems to work now without a version string.

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**